core/src/main/scala-2.13+/cats/data/NonEmptyLazyList.scala

```diff
@@ -345,14 +345,17 @@ class NonEmptyLazyListOps[A](private val value: NonEmptyLazyList[A])
   /**
    * Remove duplicates. Duplicates are checked using `Order[_]` instance.
    */
-  def distinct[AA >: A](implicit O: Order[AA]): NonEmptyLazyList[AA] = {
-    implicit val ord: Ordering[AA] = O.toOrdering
+  override def distinct[AA >: A](implicit O: Order[AA]): NonEmptyLazyList[AA] = distinctBy(identity[AA])
+
+  override def distinctBy[AA >: A, B](f: A => B)(implicit O: Order[B]): NonEmptyLazyList[AA] = {
```

Comment:
I'm concerned quite a bit about this type boundary: `AA >: A`.
It doesn't seem doing any useful work here.

Although the former method `distinct` requires that type boundary to overcome a variance error, e.g.:
> covariant type A occurs in invariant position in type cats.Order[A] of value O

or something like that.

However, in `distinctBy` it doesn't seem necessary and can only make calling `distinctBy` more difficult in some specific cases.

Reply:
The same concern applies to the other `distinctBy` method down below.

Reply:
> Although the former method distinct requires that type boundary to overcome a variance error,

Just for a record, I feel the widening of the result type in the original `distinct` implementation does not seem necessary either. I believe, it could be just this:
```scala
def distinct[AA >: A](implicit O: Order[AA]): NonEmptyLazyList[A] // <- can return `A`, not `AA`
```

Reply:
Also just for a record, looking at the original implementation I find it far from efficient: it uses `foldLeft` which is not really necessary there. Especially when results get accumulated in a mutable collection (`ListBuffer` in that case). I think it would be more efficient to use a regular loop there to avoid creation of anonymous functions and closures.

Reply:
> I'm concerned quite a bit about this type boundary: `AA >: A`. It doesn't seem doing any useful work here.
> 
> Although the former method `distinct` requires that type boundary to overcome a variance error, e.g.:
> 
> > covariant type A occurs in invariant position in type cats.Order[A] of value O
> 
> or something like that.
> 
> However, in `distinctBy` it doesn't seem necessary and can only make calling `distinctBy` more difficult in some specific cases.

Thanks for your suggestion. I've simplified the type signature for `distinctBy`, as per your suggestion. 
`distinct` is unchanged.